packages/rpc-types/src/typed-numbers.ts

```diff
@@ -1,8 +1,10 @@
+import { Brand } from '@solana/nominal-types';
+
 export type Slot = bigint;
 export type Epoch = bigint;
 
 // Specifically being used to denote micro-lamports, which are 0.000001 lamports.
-export type MicroLamports = bigint & { readonly __brand: unique symbol };
+export type MicroLamports = Brand<bigint, 'MicroLamports'>;
 export type SignedLamports = bigint;
```

Comment:
(I don't know what SignedLamports are) -- why does it not use the Lamports type above?

Should Slot and Epoch also have branded types?

Reply:
We created this type because the value of lamports in [`Rewards`](https://github.com/anza-xyz/kit/blob/2f4b5115adf7ff4c2669685a343d9b41f67d7571/packages/rpc-types/src/transaction.ts#L376-L383) could be negative. Wanted to drive home that this value was not unsigned.

I think it's important to make things like `Lamports` branded types so that you can't accidentally supply a number denominated in SOL when the API requires it in `Lamports` (as folks have been known to do). Slots and epochs are perhaps less important to brand, because they're not ‘denominations’ that are easily confused in the same way that SOL and `Lamports` both represent the same store of value, denominated in different ways.